packages/thirdweb/src/wallets/aws-kms.ts

```diff
@@ -0,0 +1,129 @@
+import type { KMSClientConfig } from "@aws-sdk/client-kms";
+import { KmsSigner } from "aws-kms-signer";
+import type {
+  SignableMessage,
+  TransactionSerializable,
+  TypedData,
+  TypedDataDefinition,
+} from "viem";
+import { hashTypedData, toBytes } from "viem";
+import { getCachedChain } from "../chains/utils.js";
+import type { ThirdwebClient } from "../client/client.js";
+import { eth_sendRawTransaction } from "../rpc/actions/eth_sendRawTransaction.js";
+import { getRpcClient } from "../rpc/rpc.js";
+import { serializeTransaction } from "../transaction/serialize-transaction.js";
+import type { Address } from "../utils/address.js";
+import type { Hex } from "../utils/encoding/hex.js";
+import { keccak256 } from "../utils/hashing/keccak256.js";
+import type { Account } from "./interfaces/wallet.js";
+
+const Buffer = globalThis.Buffer;
+
+type SendTransactionResult = {
+  transactionHash: Hex;
+};
+
+type SendTransactionOption = TransactionSerializable & {
+  chainId: number;
+};
+
+type AwsKmsAccountOptions = {
+  keyId: string;
+  config?: KMSClientConfig;
+  client: ThirdwebClient;
+};
+
+export async function getAwsKmsAccount(
+  options: AwsKmsAccountOptions,
+): Promise<Account> {
+  if (typeof Buffer === "undefined") {
```

Comment:
because we don't polyfill Buffer, and we don't expect folks to use AWS KMS on browsers, I have explicitly safeguarded against browser usage